nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/RecordStats.java

```diff
@@ -0,0 +1,225 @@
+        .build();
+
+    static final PropertyDescriptor LIMIT = new PropertyDescriptor.Builder()
+        .name("record-stats-limit")
+        .description("Limit the number of individual stats that are returned for each record path to the top N results.")
+        .required(true)
+        .defaultValue("10")
+        .addValidator(StandardValidators.INTEGER_VALIDATOR)
+        .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
+        .build();
+
+    static final Relationship REL_SUCCESS = new Relationship.Builder()
+        .name("success")
+        .description("If a flowfile is successfully processed, it goes here.")
+        .build();
+    static final Relationship REL_FAILURE = new Relationship.Builder()
+        .name("failure")
+        .description("If a flowfile fails to be processed, it goes here.")
+        .build();
+
+    private RecordPathCache cache;
+
+    static final Set RELATIONSHIPS;
+    static final List<PropertyDescriptor> PROPERTIES;
+
+    static {
+        Set _rels = new HashSet();
+        _rels.add(REL_SUCCESS);
+        _rels.add(REL_FAILURE);
+        RELATIONSHIPS = Collections.unmodifiableSet(_rels);
+        List<PropertyDescriptor> _temp = new ArrayList<>();
+        _temp.add(RECORD_READER);
+        _temp.add(LIMIT);
+        PROPERTIES = Collections.unmodifiableList(_temp);
+    }
+
+    protected PropertyDescriptor getSupportedDynamicPropertyDescriptor(final String propertyDescriptorName) {
+        return new PropertyDescriptor.Builder()
+            .name(propertyDescriptorName)
+            .displayName(propertyDescriptorName)
+            .dynamic(true)
+            .addValidator(StandardValidators.NON_BLANK_VALIDATOR)
+            .build();
+    }
+
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return PROPERTIES;
+    }
+
+    @OnScheduled
+    public void onEnabled(ProcessContext context) {
+        cache = new RecordPathCache(25);
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return RELATIONSHIPS;
+    }
+
+    @Override
+    public void onTrigger(ProcessContext context, ProcessSession session) throws ProcessException {
+        FlowFile input = session.get();
+        if (input == null) {
+            return;
+        }
+
+        try {
+            Map<String, RecordPath> paths = getRecordPaths(context);
+            Map<String, String> stats = getStats(input, paths, context, session);
+
+            input = session.putAllAttributes(input, stats);
+
+            session.transfer(input, REL_SUCCESS);
+
+        } catch (Exception ex) {
+            getLogger().error("Error processing stats.", ex);
+            session.transfer(input, REL_FAILURE);
+        }
+
+    }
+
+    protected Map<String, RecordPath> getRecordPaths(ProcessContext context) {
+        return context.getProperties().keySet()
+            .stream().filter(p -> p.isDynamic() && !p.getName().contains(RECORD_READER.getName()))
```

Comment:
I think `!p.getName().contains(RECORD_READER.getName())` part is not necessary.

Reply:
Done.